src/targets/gpu/kernels/include/migraphx/kernels/index.hpp

```diff
@@ -130,6 +130,8 @@ struct index
         return blockDim.x;
     }
 #endif
+
+    constexpr auto ngroup() const { return nglobal() / max_nlocal(); }
```

Comment:
This will usually be an integral constant, whereas `gridDim.x` is a runtime value. `for_stride` uses the compile-time constants to better optimize the strided loop.